enterprise_access/apps/content_assignments/api.py

```diff
@@ -200,19 +201,40 @@ def get_assignment_for_learner(
         constraint across [assignment_configuration,lms_user_id,content_key].  BUT still technically possible if
         internal staff managed to create a duplicate assignment configuration for a single enterprise.
     """
+    queryset = LearnerContentAssignment.objects.select_related('assignment_configuration')
+
+    try:
+        # First, try to find a corresponding assignment based on the specific content key,
+        # considering both assignments' content key and parent content key.
+        return queryset.get(
+            Q(content_key=content_key) | Q(parent_content_key=content_key),
+            assignment_configuration=assignment_configuration,
+            lms_user_id=lms_user_id,
+        )
```

Comment:
For completeness, should we handle-log-raise the "multiple records returned" exception too?  I'm imagining the only time we'd see this is if there's simultaneously a legacy and new assignment for the same course & customer, which shouldn't happen because our business logic should prevent it. That said, explicitly handling the exception might answer any questions from future code readers about why it's not handled.

Reply:
@pwnage101 Updated to include an `except` block as follows:

```python
except LearnerContentAssignment.MultipleObjectsReturned as exc:
  logger.error(
    f'Multiple assignments found with content_key or parent_content_key {content_key} '
    f'for {assignment_configuration} and lms_user_id {lms_user_id}',
  )
  raise exc
```